addons/account/models/chart_template.py

```diff
@@ -306,6 +306,10 @@ def tax_template_changed(tax, template):
                                 'record': existing_account,
                                 'noupdate': True,
                             }])
+                    # preserve reconcile value, or it may trigger Exceptions in _toggle_reconcile_to_false()
+                    account = self.ref(xmlid, raise_if_not_found=False)
```

Comment:
This will be a different record than the one returned on line 297 if and when the xmlid has been updated on line 304.

Reply:
Ah yes right.
I ask for so to do the final review & r+ ?

Reply:
Just a 
```python
if account or existing_account:
    data[model_name][xmlid].pop('reconcile', None)
```
should be enough

Reply:
This does not do what we want. This way, reconcile is set to False when it shouldn't be. In the DB upgrade I'm looking at, the exception is raised for the account.account record with id 353:
Before the upgrade:
```
$ echo "select reconcile from account_account where id=353;" | psql cawo_775683
 reconcile 
-----------
 t
```
After the upgrade:
```
$ echo "select reconcile from account_account where id=353;" | psql cawo_775683_saas~16.2 
 reconcile 
-----------
 f
```
and still:
```
$ echo "select count(*) FROM account_move_line where account_id='353' and full_reconcile_id is null and company_id=1;" | psql cawo_775683_saas~16.2 
 count 
-------
   144
```
So this way the record is updated with "false" even though that should not be allowed, because it has 144 associated move lines where  full_reconcile_id is NULL.

Reply:
Would you mind explaining why the value of `reconcile` changes?
I must be missing something obvious

Reply:
> Would you mind explaining why the value of `reconcile` changes? I must be missing something obvious

This took a while, but I can now explain it. When a field is written, the ORM will walk over a `TriggerTree` to determine necessary recomputes based on field dependencies. Before doing that, though, it will determine a list of fields that are to be "protected" from recompute based on some criteria that also match the `reconcile` field. But, it will only look at fields at all that appear in the data to be written. So when you remove the `"reconcile"` key from `data[model_name][xmlid]`, the `reconcile` field will still be part of the dependency tree, but will not be in the list of protected fields. Thus, with my proposal no recompute of `reconcile` happens, while with yours it does.

Now ... is the recompute acceptable or not? My gut feeling is that it seems wrong during an upgrade, but please correct me if I'm wrong.

![func-protecting-cawo](https://github.com/odoo/odoo/assets/130560545/303d2e78-5b8a-4e9e-9c83-f482aa178aa4)

![funx-protecting-wan](https://github.com/odoo/odoo/assets/130560545/4dfc7df0-36b2-45f3-8c0c-dcedecec1a44)

Reply:
One other thought ... If and when it is incorrect to set reconcile=False on that record on one code-path, I would expect it to be wrong on the other (in _compute_reconcile), too. If that assumption is correct, then this would be a strong argument to replace the python constraint by a partial SQL constraint.

Reply:
Ok, so it is because writing on `account_type` triggers the recomputation
I would argue only keeping the `tag_ids` for accounts that already exist in that case.

Anyways, there is no need to redo a `ref` :wink: (lets avoid useless queries if possible)

Reply:
> Ok, so it is because writing on `account_type` triggers the recomputation

yes

> I would argue only keeping the `tag_ids` for accounts that already exist in that case.

You mean tag_ids should be preserved, too? Sounds good. I can add that.

> Anyways, there is no need to redo a `ref` wink (lets avoid useless queries if possible)

https://github.com/odoo/odoo/pull/125320#discussion_r1232134184

Reply:
> Quite the opposite. Everything should be preserved on existing accounts except the tags.

Oh, OK. Understood. How about the latest version, then?